internal/planner/telemetry.go

```diff
@@ -30,38 +37,106 @@ func must[T any](obj T, err error) T {
 }
 
 var (
-	pkg = "github.com/canonical/github-runner-operators/internal/planner"
+	pkg    = "github.com/canonical/github-runner-operators/internal/planner"
+	logger = telemetry.NewLogger(pkg)
+	trace  = otel.Tracer(pkg)
 )
 
 // Metrics encapsulates all OpenTelemetry metrics for the planner service.
 type Metrics struct {
-	mu                   sync.RWMutex
-	store                FlavorStore
-	meter                metric.Meter
-	logger               *slog.Logger
-	flavorPressureMetric metric.Int64ObservableGauge
+	mu    sync.RWMutex
+	store FlavorStore
+	meter metric.Meter
+
+	flavorPressure        metric.Int64ObservableGauge
+	webhookErrors         metric.Int64Counter
+	processedWebhooks     metric.Int64Counter
+	discardedWebhooks     metric.Int64Counter
+	webhookJobWaitingTime metric.Float64Histogram
+	webhookJobRunningTime metric.Float64Histogram
 }
 
 // NewMetrics initializes the planner metrics and registers the observable gauge.
 func NewMetrics(store FlavorStore) *Metrics {
 	m := &Metrics{
-		store:  store,
-		meter:  otel.Meter(pkg),
-		logger: telemetry.NewLogger(pkg),
+		store: store,
+		meter: otel.Meter(pkg),
 	}
 
-	m.flavorPressureMetric = must(
+	m.flavorPressure = must(
 		m.meter.Int64ObservableGauge(
 			flavorPressureMetricName,
 			metric.WithDescription("The current pressure value for each flavor"),
 			metric.WithUnit("{pressure}"),
 		),
 	)
+	m.webhookErrors = must(
+		m.meter.Int64Counter(
+			webhookErrorsMetricName,
+			metric.WithDescription("Total number of webhook processing errors"),
+			metric.WithUnit("{error}"),
+		),
+	)
+	m.discardedWebhooks = must(
+		m.meter.Int64Counter(
+			discardedWebhooksMetricName,
+			metric.WithDescription("Total number of discarded webhooks"),
+			metric.WithUnit("{webhook}"),
+		),
+	)
+	m.processedWebhooks = must(
+		m.meter.Int64Counter(
+			consumedWebhooksMetricName,
+			metric.WithDescription("Total number of processed received"),
+			metric.WithUnit("{webhook}"),
+		),
+	)
+	jobDurationBucket := metric.WithExplicitBucketBoundaries(
```

Comment:
nit:
```suggestion
	jobDurationOrWaitingBucket := metric.WithExplicitBucketBoundaries(
```

or better defining a second variable `jobWaiting := jobDuration`. Allows faster human reading, fewer interpretation errors

Reply:
Updated to use two variables, thanks!